File: app/src/main/java/org/michaelbel/moviemade/extensions/DeviceUtil.java
package org.michaelbel.moviemade.extensions;

import android.content.Context;
import android.content.res.Configuration;

import org.michaelbel.moviemade.R;

import androidx.annotation.NonNull;

public class DeviceUtil {

    public static boolean isTablet(@NonNull Context context) {
        return context.getResources().getBoolean(R.bool.tablet);
    }

    public static boolean isLandscape(@NonNull Context context) {
        return context.getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE;
    }

    public static int getStatusBarHeight(@NonNull Context context) {
        int result = 0;
        int resId = context.getResources().getIdentifier("status_bar_height", "dimen", "android");

        if (resId > 0) {
            result = context.getResources().getDimensionPixelSize(resId);
        }

        return result;
    }

    public static int dp(@NonNull Context context, float value) {
        return (int) Math.ceil(context.getResources().getDisplayMetrics().density * value);
    }
}

File: app/src/main/java/org/michaelbel/moviemade/extensions/DeviceUtil.kt
package org.michaelbel.moviemade.extensions

import android.content.Context
import android.content.res.Configuration
import org.michaelbel.moviemade.R

object DeviceUtil {

    fun isTablet(context: Context): Boolean {
        return context.resources.getBoolean(R.bool.tablet)
    }

    fun isLandscape(context: Context): Boolean {
        return context.resources.configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
    }

    fun getStatusBarHeight(context: Context): Int {
        var result = 0
        val resId = context.resources.getIdentifier("status_bar_height", "dimen", "android")

        if (resId > 0) {
            result = context.resources.getDimensionPixelSize(resId)
        }

        return result
    }

    fun dp(context: Context, value: Float): Int {
        return Math.ceil((context.resources.displayMetrics.density * value).toDouble()).toInt()
    }
}